### PR TITLE
yandex-music: 5.18.2 -> 5.23.2

### PR DESCRIPTION
**Pull request by ein-shved**
Fix description according to nixpkgs format.
    
Reuse version number in changelog field url.
    
Drop creation of extra info.json file.
    
Do not save "electron" section of version_info.json file from original
repository.
    
Use SRI hash for exe file instead of hex. Make update script to convert
from hex to SRI format.

yandex-music: 5.18.2 -> 5.23.2

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 350288` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>yandex-music</li>
  </ul>
</details>

File: pkgs/by-name/ya/yandex-music/info.json
Comment (dotlambda) [inline pkgs/by-name/ya/yandex-music/info.json:1]: Why does this file exist? It isn't used.

File: pkgs/by-name/ya/yandex-music/ym_info.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-    "version": "5.18.2",
-    "exe_name": "Yandex_Music_x64_5.18.2.exe",
-    "exe_link": "https://music-desktop-application.s3.yandex.net/stable/Yandex_Music_x64_5.18.2.exe",
-    "exe_sha256": "a513eb6aa4bd05bae8503f5578a8e91d778794ffe4eb3a450040903570106941"
+    "version": "5.23.2",
+    "exe_name": "Yandex_Music_x64_5.23.2.exe",
+    "exe_link": "https://music-desktop-application.s3.yandex.net/stable/Yandex_Music_x64_5.23.2.exe",
+    "exe_sha256": "3e15e650657dfbdcdf459b9fc37155be93c6a952c0079d65af3eca268be6f82f"
   },
   "electron": {
```
Comment (dotlambda) [inline pkgs/by-name/ya/yandex-music/ym_info.json:8]: This section doesn't seem to be used.

File: pkgs/by-name/ya/yandex-music/info.json
Comment (ein-shved) [reply pkgs/by-name/ya/yandex-music/info.json:1]: Yep. It was developing artifact. Sorry

File: pkgs/by-name/ya/yandex-music/ym_info.json (hunk 1 above)
Comment (ein-shved) [reply pkgs/by-name/ya/yandex-music/ym_info.json]: Yes, this file downloaded from original repository as is by updater script. I've modified  script to drop this section.

File: pkgs/by-name/ya/yandex-music/package.nix [hunk 2]
```diff
@@ -38,8 +38,8 @@ stdenvNoCC.mkDerivation rec {
       ym_info = builtins.fromJSON (builtins.readFile ./ym_info.json);
     in
     fetchurl {
-      url = ym_info.ym.exe_link;
-      sha256 = ym_info.ym.exe_sha256;
+      url = ym_info.exe_link;
+      sha256 = ym_info.exe_sha256;
```
Comment (dotlambda) [inline pkgs/by-name/ya/yandex-music/package.nix:42]:
```suggestion
      hash = ym_info.exe_sha256;
```

File: pkgs/by-name/ya/yandex-music/ym_info.json [hunk 3]
```diff
@@ -1,14 +1,6 @@
...
-  "electron": {
-    "version": "29.4.6",
-    "x64": "https://github.com/electron/electron/releases/download/v29.4.6/electron-v29.4.6-linux-x64.zip",
-    "armv7l": "https://github.com/electron/electron/releases/download/v29.4.6/electron-v29.4.6-linux-armv7l.zip",
-    "arm64": "https://github.com/electron/electron/releases/download/v29.4.6/electron-v29.4.6-linux-arm64.zip"
-  }
+  "version": "5.23.2",
+  "exe_name": "Yandex_Music_x64_5.23.2.exe",
+  "exe_link": "https://music-desktop-application.s3.yandex.net/stable/Yandex_Music_x64_5.23.2.exe",
+  "exe_sha256": "3e15e650657dfbdcdf459b9fc37155be93c6a952c0079d65af3eca268be6f82f"
```
Comment (dotlambda) [inline pkgs/by-name/ya/yandex-music/ym_info.json:5]: Use an SRI hash.

Comment (ein-shved) [reply pkgs/by-name/ya/yandex-music/ym_info.json:5]: I can not change this in original project. This file is auto-generated from [this](https://github.com/cucumber-sp/yandex-music-linux/blob/master/utility/version_info.json)

Comment (ein-shved) [reply pkgs/by-name/ya/yandex-music/ym_info.json]: Okay! I've googled a bit and found the way to convert. Add conversion to unpdateScript

**Comment by ein-shved** [comment]
@dotlambda Hi! Do you have any more comments to this PR?

**Pull request merged**